examples/typescript/swap.ts

```diff
@@ -0,0 +1,275 @@
+/**
+ * Example to demonstrate creating and adding to liquidity pools, swapping between two fungible asset.
+ *
+ * Steps:
+ * 1. Create two accounts (Alice and Bob)
+ * 2. Fund Alice and Bob
+ * 3. Create a FA with Alice as the owner
+ * 4. Create another FA with Bob as the owner
+ * 5. Create a liquidity pool with Alice's FA and Bob's FA
+ * 6. Swap between Alice's FA and Bob's FA
+ */
+
+import {
+  Account,
+  AccountAddress,
+  Aptos,
+  Bool,
+  Ed25519PrivateKey,
+  Hex,
+  HexInput,
+  MoveObject,
+  MoveString,
+  U64,
+  ViewRequestData,
+} from "aptos";
+
+const readline = require("readline").createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const getOptimalLpAmount = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  token1Addr: HexInput,
+  token2Addr: HexInput,
+): Promise<void> => {
+  const payload: ViewRequestData = {
+    function: `0x${swap.toStringWithoutPrefix()}::router::optimal_liquidity_amounts`,
+    arguments: [
+      `0x${Hex.fromHexInput(token1Addr).toStringWithoutPrefix()}`,
+      `0x${Hex.fromHexInput(token2Addr).toStringWithoutPrefix()}`,
+      false,
+      "200000",
+      "300000",
+      "200",
+      "300",
+    ],
+  };
+  const result: any = await aptos.view({ payload });
+  console.log("Optimal LP amount: ", result);
+};
+
+const addLiquidity = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  deployer: Account,
+  token1Addr: HexInput,
+  token2Addr: HexInput,
+): Promise<string> => {
+  const rawTxn = await aptos.generateTransaction({
+    sender: deployer.accountAddress.toString(),
+    data: {
+      function: `0x${swap.toStringLongWithoutPrefix()}::router::add_liquidity_entry`,
```

Comment:
I think the 0 prefix should still work? toStringLongWithoutPrefix is a bit long instead of toString

Reply:
it is not about what will work or not. it is about the strict type we hold in the sdk. we ask the `function` to be of type `0x${string}::${string}::${string}` but since `swap.toString()` is just a string (tho it does have 0x) typescript will complain that type dont match. 
It is just something we need to change in the sdk

Reply:
Fixed now.